test/archethic/self_repair/sync/transaction_handler_test.exs

```diff
@@ -140,23 +145,81 @@ defmodule Archethic.SelfRepair.Sync.TransactionHandlerTest do
       }
     ]
 
+    welcome_node_pkey = welcome_node.first_public_key
+    coordinator_node_pkey = coordinator_node.first_public_key
+    storage_node_pkey = Enum.at(storage_nodes, 0).first_public_key
+
     tx = TransactionFactory.create_valid_transaction(inputs)
 
-    modified_tx = %{tx | type: :oracle}
+    MockClient
+    |> stub(:send_message, fn
+      %Node{first_public_key: ^welcome_node_pkey}, %GetTransaction{}, _ ->
+        {:ok, tx}
+
+      %Node{first_public_key: ^coordinator_node_pkey}, %GetTransaction{}, _ ->
+        # split
+        {:ok,
+         put_in(tx, [Access.key!(:validation_stamp), Access.key!(:timestamp)], DateTime.utc_now())}
```

Comment:
Since TransactionFactory also create transaction using `DateTime.utc_now()` as validation time, if your test run in the same millisecond, the transaction will be the same.
To avoid wrong or flaky test you should update something constant like transaction type or content